File: code/src/test/scala/cern/colt/matrix/tint/impl/DiagonalIntMatrix2DViewTest.scala
package cern.colt.matrix.tint.impl

//remove if not needed
import scala.collection.JavaConversions._

class DiagonalIntMatrix2DViewTest(arg0: String) extends DiagonalIntMatrix2DTest(arg0) {

  protected def createMatrices() {
    DINDEX = 3
    A = new DiagonalIntMatrix2D(NCOLUMNS, NROWS, -DINDEX)
    DLENGTH = A.asInstanceOf[DiagonalIntMatrix2D].diagonalLength()
    A = A.viewDice()
    B = new DiagonalIntMatrix2D(NCOLUMNS, NROWS, -DINDEX).viewDice()
    Bt = new DiagonalIntMatrix2D(NROWS, NCOLUMNS, DINDEX).viewDice()
  }
}

File: src/test/java/cernx/colt/matrix/tint/impl/DiagonalIntMatrix2DViewTest.java
package cernx.colt.matrix.tint.impl;

public class DiagonalIntMatrix2DViewTest extends DiagonalIntMatrix2DTest {

    public DiagonalIntMatrix2DViewTest(String arg0) {
        super(arg0);
    }

    protected void createMatrices() throws Exception {
        DINDEX = 3;
        A = new DiagonalIntMatrix2D(NCOLUMNS, NROWS, -DINDEX);
        DLENGTH = ((DiagonalIntMatrix2D) A).diagonalLength();
        A = A.viewTranspose();
        B = new DiagonalIntMatrix2D(NCOLUMNS, NROWS, -DINDEX).viewTranspose();
        Bt = new DiagonalIntMatrix2D(NROWS, NCOLUMNS, DINDEX).viewTranspose();
    }

}
